programs/ssl/ssl_server2.c

```diff
@@ -1620,8 +1620,10 @@ int main( int argc, char *argv[] )
 #if !defined(_WIN32)
     if( received_sigterm )
     {
-        polarssl_printf( " interrupted by SIGTERM\n" );
-        ret = 0;
+        polarssl_printf( " interrupted by SIGTERM (not in net_accept())\n" );
+        if( ret == POLARSSL_ERR_NET_RECV_FAILED )
```

Comment:
This could be `SEND_FAILED` as well as `RECV_FAILED`.

Annoyingly there's no `ssl-opt.sh -p` to check what happens in the nominal case, but it's easy enough to hard-code it:
```
--- a/tests/ssl-opt.sh
+++ b/tests/ssl-opt.sh
@@ -371,7 +371,9 @@ run_test() {
 
     # if we're here, everything is ok
     echo "PASS"
-    rm -f $SRV_OUT $CLI_OUT
+    #rm -f $SRV_OUT $CLI_OUT
+    mv $SRV_OUT o-srv-${TESTS}.log
+    mv $CLI_OUT o-cli-${TESTS}.log
 }
```
I did that, checked the logs, and saw a few instances of `NET_FAILED` and no other error apart from the ones suppressed by https://github.com/ARMmbed/mbedtls/pull/1291/commits/c346ddbaff461589eda71e0adf591037fd8e9137.

Reply:
Aw good catch! I'll fix by adding `SEND_FAILED` here.